### DISCUSSION TITLE
Possible Feature: Article Series

### DISCUSSION THREAD

**Discussion by kera-cudmore** [codu-code/codu#579] [Ideas]
- Add feature that would allow you to mark articles as part of a series.

This is a feature that they have on dev.to, which allows a user to mark their article as part of a series. This then provides a list at the beginning of the articles letting the user know it is part of a series and allows another way for the user to view those articles directly by clicking on the list

![Screenshot 2023-04-17 at 2 14 10 pm](https://user-images.githubusercontent.com/92253071/232494995-22dcdc0d-4695-438d-b937-57b45d589be8.png)
![Screenshot 2023-04-17 at 2 13 50 pm](https://user-images.githubusercontent.com/92253071/232494999-391c037d-405c-4d68-aa97-9aa3811251e1.png)

**Comment by NiallJoeMaher** [comment]
I love this so I'll take a look into how we can do it.

**Comment by JohnAllenTech** [comment]
Would you just add a FK value to the article schema?

FK of a new entry on a "article series" table and then the position on of the article could be stored in the series entry

Something like this maybe?
<img width="910" alt="image" src="https://github.com/codu-code/codu/assets/46611809/5947ebcd-2f11-4350-bfc0-20f058eaafe8">

Not a finished though but maybe it will spark a discussion on how this can be done

**Comment by CBID2** [comment]
Hi @JohnAllenTech! 🙂 I'm @CBID2, nice to meet you! 🫱🏾‍🫲🏻Did you make any progress in creating this feature?

**Comment by NiallJoeMaher** [comment]
@CBID2, I don't think there has been progress yet! We were chatting in the Discord about it but I'm not sure if there was a decision on how to tackle it.

**Comment by CBID2** [comment]
> @CBID2, I don't think there has been progress yet! We were chatting in the Discord about it but I'm not sure if there was a decision on how to tackle it. 

Ok keep me posted @NiallJoeMaher! I want to help create a tutorial for it once it's published

**Comment by JohnAllenTech** [comment]
@NiallJoeMaher 

Did you get a chance to look at this?
![image](https://github.com/codu-code/codu/assets/46611809/ee0423a8-579d-4450-8f31-98369dd17760)

Only small snag I see is order of the articles within the series. Storing that in the article series table makes the most sense to me as otherwise it will require all multiple article rows to be updated when a sign article is moved.

If this looks good from a technical perspective can we break this ticket into a FE and BE portion to make it more managable, I can take the BE and maybe if we can get a design for the page someone else can grab the FE

**Comment by JohnAllenTech** [comment]
@NiallJoeMaher are you ok with me taking this and running with it?

**Comment by NiallJoeMaher** [comment]
@JohnAllenTech yes sir! Do you mind taking a look at it as we chatted in Discord? Where we add a `Series` table ad  `series_id` to the `Post` table.

As a rough idea I'm thinking the `series` table might have some of the following details:
`id`, `title`, `description`, `userId`, `createdAt`, and `updatedAt`.

We can order the articles from the `published` date. 

But I'm happy to see your own suggestions/improvements. I'm just pulling these based on Discord. 🦾

**Comment by JohnAllenTech** [comment]
Sure Apologies I was away on holibops for the last two weeks so I am only taking a look at this now. Will have an update soon enough

**Comment by CBID2** [comment]
> Sure Apologies I was away on holibops for the last two weeks so I am only taking a look at this now. Will have an update soon enough

It's all good @JohnAllenTech! :) I can't wait to see how this grows.

**Comment by JohnAllenTech** [comment]
Another update on this.

have the article table added to Postgres and migration seems to be working. Couple of snags but on the other side of them now.

Hoping to share some screenshots and gifs of a rough version of this by early next week

**Comment by CBID2** [comment]
> Another update on this.
> 
> 
> 
> have the article table added to Postgres and migration seems to be working. Couple of snags but on the other side of them now.
> 
> 
> 
> Hoping to share some screenshots and gifs of a rough version of this by early next week

Keep it up @JohnAllenTech! :)

**Comment by JohnAllenTech** [comment]
Still chugging on this. Really busy between work and some volunteering commitments.

Starting work on the article series page next. The plumbing to get to the article series page is working and the series is in the DB.

Any design inspiration is greatly received

**Comment by JohnAllenTech** [comment]
Article series page
Will need design help to alter the preview of the series to be different then an article
<img width="1426" alt="image" src="https://github.com/codu-code/codu/assets/46611809/c6632f47-f403-456a-9093-5ea6798addb4">


Article page
Got it routing correctly a few minute ago so just a blank canvas for the time being but I have all the data to display articles in this series so gonna hit that first
<img width="1426" alt="image" src="https://github.com/codu-code/codu/assets/46611809/09ddb73d-899d-409f-9ac9-73e7e77dfcaa">

**Comment by CBID2** [comment]
> Still chugging on this. Really busy between work and some volunteering commitments.
> 
> Starting work on the article series page next. The plumbing to get to the article series page is working and the series is in the DB.
> 
> Any design inspiration is greatly received

Have you checked out Hashnode @JohnAllenTech?

**Comment by JohnAllenTech** [comment]
@CBID2 for design inspo? No I haven’t will take a look. Thanks

**Comment by NiallJoeMaher** [comment]
This one has gone stale but I would love to revive it! It seems like a perfect Hacktoberfest feature we could all collab on.

**Comment by CBID2** [comment]
> This one has gone stale but I would love to revive it! It seems like a perfect Hacktoberfest feature we could all collab on. 

I'm still up to do a tutorial on that once this feature is finished @NiallJoeMaher! :)

**Comment by JohnAllenTech** [comment]
Apologies all. Just can’t find the time for any OSS work at the minute.

**Comment by NiallJoeMaher** [comment]
Moved this to the discussion tab so I can open specific issues for this one. 🦾
